File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/universal/driver/TCookie.java
package com.applitools.eyes.playwright.universal.driver;

import com.applitools.eyes.universal.driver.ICookie;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.microsoft.playwright.options.Cookie;

public class TCookie extends Cookie implements ICookie {

    private Double expiry;

    public TCookie(Cookie cookie) {
        this(cookie.name, cookie.value);
        setExpires(cookie.expires);
        setDomain(cookie.domain);
        setPath(cookie.path);
        setUrl(cookie.url);
        setSecure(cookie.secure);
        setHttpOnly(cookie.httpOnly);
        setSameSite(cookie.sameSite);
    }

    private TCookie(String name, String value) {
        super(name, value);
    }

    public Double getExpires() {
        return expiry;
    }

    @JsonProperty("expiry")
    @Override
    public TCookie setExpires(double expires) {
        this.expiry = expires;
        return this;
    }
}

File: dotnet/Eyes.Playwright.DotNet/Universal/Driver/TCookie.cs
using Applitools.Universal.Driver;
using Microsoft.Playwright;

namespace Applitools.Playwright.Universal.Driver
{
    public class TCookie : Cookie, ICookie
    {
        public TCookie(BrowserContextCookiesResult cookie)
            : this(cookie.Name, cookie.Value)
        {
            Expires = cookie.Expires;
            Domain = cookie.Domain;
            Path = cookie.Path;
            Secure = cookie.Secure;
            HttpOnly = cookie.HttpOnly;
            SameSite = cookie.SameSite;
        }

        public TCookie(Cookie cookie)
            : this(cookie.Name, cookie.Value)
        {
            Expires = cookie.Expires;
            Domain = cookie.Domain;
            Path = cookie.Path;
            Url = cookie.Url;
            Secure = cookie.Secure;
            HttpOnly = cookie.HttpOnly;
            SameSite = cookie.SameSite;
        }

        private TCookie(string name, string value)
        {
            Name = name;
            Value = value;
        }
    }
}